src/items.zig

```diff
@@ -353,11 +345,24 @@ const ToolPhysics = struct { // MARK: ToolPhysics
 		}
 		var tempModifiers: main.List(Modifier) = .init(main.stackAllocator);
 		defer tempModifiers.deinit();
+		for(tool.type.properties) |property| {
+			var sum: f32 = 0;
+			var weight: f32 = 0;
+			for(0..25) |i| {
+				const material = (tool.craftingGrid[i] orelse continue).material orelse continue;
+				sum += property.weigths[i]*material.getProperty(property.source);
+				weight += property.weigths[i];
+			}
+			if(weight == 0) continue;
+			sum /= weight;
+			sum *= property.resultScale;
+			tool.getProperty(property.destination).* += sum;
+		}
+		if(tool.maxDurability < 1) tool.maxDurability = 1;
+		if(tool.damage < 1) tool.damage = 1/(2 - tool.damage);
+		if(tool.swingTime < 1) tool.swingTime = 1/(2 - tool.swingTime);
```

Comment:
Hmm having same thing for swing time means that to get swing time of 0.23s (current iron pickaxe) requires swing time value to be like -7, while this is not necessarily unhealthy, its a bit strange, how do we explain that intuitively?

Reply:
Problem solved.